File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/interfaces/services/ILyricsSearchService.kt
package com.jayfunc.betterlyrics.interfaces.services
interface ILyricsSearchService {
    suspend fun searchSmartlyAsync(songInfo: Any, token: Any): Any?
    suspend fun searchAllAsync(songInfo: Any, token: Any): List<Any>
    fun getActiveProviders(): List<Any>
    suspend fun downloadAmllTtmlDbIndexAsync(token: Any)
}

File: src/BetterLyrics.DotNet/BetterLyrics.Core/Interfaces/Services/ILyricsSearchService.cs
// 2025/6/23 by Zhe Fang

using BetterLyrics.Core.Enums;
using BetterLyrics.Core.Models;
using BetterLyrics.Core.Models.Entities;

namespace BetterLyrics.Core.Interfaces.Services;

public interface ILyricsSearchService
{
    Task<LyricsCacheItem?> SearchSmartlyAsync(SongInfo songInfo, LyricsSearchType? lyricsSearchType,
        CancellationToken token);

    IAsyncEnumerable<LyricsCacheItem> SearchAllAsync(
        SongInfo songInfo,
        bool checkCache,
        CancellationToken cancellationToken = default);

    List<LyricsProvider> GetActiveProviders();
}